core/src/main/scala/kafka/server/metadata/BrokerMetadataPublisher.scala

```diff
@@ -117,6 +117,8 @@ class BrokerMetadataPublisher(conf: KafkaConfig,
                        delta: MetadataDelta,
                        newImage: MetadataImage): Unit = {
     try {
+      debug(s"Publishing delta $delta with highest offset $newHighestMetadataOffset")
```

Comment:
Are there any concerns with the MetadataDelta having a lot of records in it? I'm wondering if this could potentially dump out a lot of text into the debug log.

Also, from a security perspective, is it okay to write the contents of _any_ metadata record to the debug log?

Reply:
I found it useful when debugging the failures here, so I thought it would come in handy when debugging system test failures as well. I guess if there were any concerns from a security perspective, it would be about the `ConfigurationDelta`. Perhaps if we want to err on the safe side, we could print the keys only? That might be good enough since it would probably be clear from the test context what the change was. What do you think?

Reply:
Yea, configs is what I was thinking of. Logging the keys only sounds like a good idea

Reply:
I decided to make this trace level since it could be a bit noisy in some cases.